colossus-docs/src/main/scala/RedisServiceExample.scala

```diff
@@ -0,0 +1,46 @@
+// #example
+import java.util.concurrent.ConcurrentHashMap
+import akka.actor.ActorSystem
+import akka.util.ByteString
+import colossus.IOSystem
+import colossus.protocols.redis._
+import colossus.protocols.redis.server.{Initializer, RedisServer, RequestHandler}
+import colossus.service.Callback
+import colossus.service.GenRequestHandler.PartialHandler
+
+object RedisServiceExample extends App {
+
+  implicit val actorSystem = ActorSystem()
+  implicit val ioSystem = IOSystem()
+
+  val db = new ConcurrentHashMap[String, String]()
+
+  RedisServer.start("example-server", 9000) {
+    new Initializer(_) {
+      override def onConnect = new RequestHandler(_) {
+        override def handle: PartialHandler[Redis] = {
+          case Command("GET", args) =>
```

Comment:
You suggesting something similar to what we have for [memcache](https://github.com/tumblr/colossus/blob/master/colossus/src/main/scala/colossus/protocols/memcache/Memcache.scala#L47)?

I didn't understand your point on macro/code-generation.